Insights/EventsController.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  EventsController.swift
+//  Insights
+//
+//  Created by Vladislav Fitc on 06/11/2018.
+//  Copyright © 2018 Algolia. All rights reserved.
+//
+
+import Foundation
+
+class EventsController: EventProcessor {
+    
+    typealias Storage = LocalStorage<[EventsPackage]>
+    
+    let credentials: Credentials
+    var eventsPackages: [EventsPackage]
+    let webservice: WebService
+    let logger: Logger
+    var isLocalStorageEnabled: Bool = true {
+        didSet {
+            if !isLocalStorageEnabled {
+                Storage.deleteFile(atPath: localStorageFileName)
+            }
+        }
+    }
+    private let dispatchQueue: DispatchQueue
+    private let localStorageFileName: String
+    private weak var flushTimer: Timer?
+    
+    init(credentials: Credentials,
+         webService: WebService,
+         flushDelay: TimeInterval,
+         logger: Logger,
+         dispatchQueue: DispatchQueue = .init(label: "insights.events", qos: .background)) {
+        self.eventsPackages = []
+        self.credentials = credentials
+        self.logger = logger
+        self.webservice = webService
+        self.dispatchQueue = dispatchQueue
+        self.localStorageFileName = "\(credentials.appId).events"
+        let flushTimer = Timer.scheduledTimer(timeInterval: flushDelay,
+                                                    target: self,
+                                                  selector: #selector(flushEvents),
+                                                  userInfo: nil,
+                                                   repeats: true)
+        self.flushTimer = flushTimer
+
+        RunLoop.main.add(flushTimer, forMode: .default)
+        deserialize()
```

Comment:
we should put more info into that. reading deserialize hides a lot of information. what is it deserializing?

- I would clarify in the name of the function what it is deserializing
- potentially, if it's mutating a property of the class (in this case `self.eventsPackage`), then for readability, I would prefer when calling the method to do `self.eventsPackage = deserialize()` or `try deserialize` or something something like that.